cdap-kms/src/main/java/co/cask/cdap/security/store/KMSTokenUtils.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright © 2015-2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.security.store;
+
+import com.google.common.base.Throwables;
+import org.apache.hadoop.crypto.key.KeyProvider;
+import org.apache.hadoop.crypto.key.KeyProviderDelegationTokenExtension;
+import org.apache.hadoop.security.Credentials;
+import org.apache.hadoop.security.UserGroupInformation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+
+/**
+ * Helper class for getting KMS security delegation token.
+ */
+public class KMSTokenUtils {
+  private static final Logger LOG = LoggerFactory.getLogger(KMSTokenUtils.class);
+
+  public static Credentials obtainToken(final KMSSecureStore secureStore, Credentials credentials) {
```

Comment:
it seems a little strange to have to go through a CDAP class (KMSSecureStore) to get the KMS token. Instead, can we pass ``Configuration`` here and construct the KMSProvider``? I know it seems a little duplicate, but because of this, you need weird casting of a ``SecureStore`` object to ``KMSSecureStore`` in ``TokenSecureStoreUpdater#107``.